src/common/debug.rs

```diff
@@ -0,0 +1,90 @@
+use std::sync::Arc;
+
+use parking_lot::Mutex;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::common::pyroscope_state::pyro::PyroscopeState;
+use crate::settings::Settings;
+
+#[derive(Serialize, JsonSchema, Debug, Deserialize, Clone)]
+pub struct PyroscopeConfig {
+    pub url: String,
+    pub identifier: String,
+    pub user: Option<String>,
+    pub password: Option<String>,
+    pub sampling_rate: Option<u32>,
+}
+
+#[derive(Default, Debug, Serialize, JsonSchema, Deserialize, Clone)]
+pub struct DebugConfig {
+    pub pyroscope: Option<PyroscopeConfig>,
+}
+
+#[derive(Debug, Serialize, JsonSchema, Deserialize, Clone)]
+#[serde(rename_all = "snake_case")]
+pub enum DebugConfigPatch {
+    Pyroscope(Option<PyroscopeConfig>),
+}
```

Comment:
Is there a reason for doing it this way with a single enum variant? Do we want to have extensibility here for other variants in the future?

Reply:
Yes. Any kind of debugger integrations (profiling, tracing, etc) in future can go here.